internal/source/local.go

```diff
@@ -0,0 +1,79 @@
+package source
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+	"github.com/operator-framework/rukpak/internal/util"
+
+	"github.com/go-git/go-billy/v5/memfs"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Local struct {
+	client.Client
+	// reader queries the API server directly
+	reader client.Reader
+}
+
+func (o *Local) Unpack(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (*Result, error) {
+	if bundle.Spec.Source.Type != rukpakv1alpha1.SourceTypeLocal {
+		return nil, fmt.Errorf("bundle source type %q not supported", bundle.Spec.Source.Type)
+	}
+	if bundle.Spec.Source.Local == nil {
+		return nil, fmt.Errorf("bundle source local configuration is unset")
+	}
+
+	configMapRef := bundle.Spec.Source.Local.ConfigMapRef
+
+	var cm corev1.ConfigMap
+	if err := o.reader.Get(ctx, client.ObjectKey{Name: configMapRef.Name, Namespace: configMapRef.Namespace}, &cm); err != nil {
+		return nil, fmt.Errorf("could not find configmap %s/%s on the cluster", configMapRef.Namespace, configMapRef.Name)
+	}
+
+	// If the configmap is empty, return early
+	if len(cm.Data) == 0 {
+		return nil, fmt.Errorf("configmap %s/%s is empty: at least one object is required", configMapRef.Namespace, configMapRef.Name)
+	}
+
+	var memFS = memfs.New()
+	for filename, contents := range cm.Data {
+		file, err := memFS.Create(filepath.Join("manifests", filename))
+		if err != nil {
+			return nil, fmt.Errorf("creating filesystem from configmap: %s", err)
+		}
+		_, err = file.Write([]byte(contents))
+		if err != nil {
+			return nil, fmt.Errorf("creating filesystem from configmap: %s", err)
+		}
+	}
+
+	// Add an ownerref to the configmap based on the Bundle
+	var ownerref = metav1.NewControllerRef(bundle, bundle.GroupVersionKind())
```

Comment:
Could you explain for me why we want Rukpak to now own the ConfigMap? In essence this means that if the Bundle is deleted then the ConfigMap gets deleted as well. Should that always be the case or would we prefer that to be configurable somehow? I can imagine a scenario where a user would not expect the ConfigMap they created to be deleted by deleting the Bundle.

Another problem that may arise from this is if two Bundles share the same ConfigMap. What happens in that scenario if we delete that ConfigMap?

Reply:
I'm looking at this for the perspective of "this Bundle binds this configmap, so tie the configmap's lifecycle to that of the Bundle". It helps keep the cluster state the same before/after a BD is applied. I think if we document the fact that the configmap is owned by the Bundle users will expect this behavior. We could make it configurable further on if we find out that users would like their configmaps to stay around. 

Good question about two Bundles sharing the same configmap -- currently that's not really supported since its implied each Bundle is backed by its own unique content source. I'm not sure I see the value of having two cluster-scoped Bundles that both reference the same backing content.

Reply:
> I'm not sure I see the value of having two cluster-scoped Bundles that both reference the same backing content.
I don't either, to be sure. I just want to make sure we've thought of it and have an official stance before putting this out.

Thanks for the answer, I feel better about this to the point I'll go ahead and mark this as resolved. Its good to document this so that it is expected.